src/DotNetWorker/Converters/ArrayConverter.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    // Converting IEnumerable<> to Array
+    internal class ArrayConverter : IConverter
+    {
+        // Convert IEnumerable to array
+        public bool TryConvert(ConverterContext context, out object? target)
+        {
+            // Ensure requested type is an array
```

Comment:
You can assign `target` to null at the beginning, which will help avoid the checks below